pkg/util/probe/readyz.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package probe
+
+import "net/http"
+
+const (
+	ReadyzEndpoint = "/readyz"
+)
+
+var (
+	ready = false
+)
+
+func SetReady() {
+	ready = true
```

Comment:
Do we need locking?
Currently SetReady() is called when the controller's go routine starts watching the cluster.etcd resource to create cluster events on the event channel.
Will there ever be two controller go routines trying to watch the cluster resource?
Also even if there were two controller go-routines, the first of them to start watching the cluster.etcd resource can safely set the state to `ready=true` since the operator would be ready by that point. We don't go back to `ready=false` ever again so there shouldn't be an issue right?

Although if you want to be safe we can just add locking.